dashboard/test/controllers/admin_users_controller_test.rb

```diff
@@ -130,4 +133,21 @@ class AdminUsersControllerTest < ActionController::TestCase
     assert_response :success
     assert_select '.container .alert-danger', 'User not found -- email not confirmed'
   end
+
+  test "undelete_user undeletes deleted user" do
+    sign_in @admin
+
+    post :undelete_user, {user_id: @deleted_student.id}
+
+    @deleted_student.reload
+    assert @deleted_student.deleted_at.nil?
+  end
+
+  test "undelete_user noops for (non-deleted) user" do
+    sign_in @admin
+
+    assert_no_difference(User.deleted_at) do
```

Comment:
ah, I guess you could manually to_i that... (nil.to_i == 0) but I guess at that point could more messier than it is cleaner :p